src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -136,6 +140,53 @@ public void Dispose()
         public DataFrame Table(string tableName) =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("table", tableName));
 
+        /// <summary>
+        /// Creates a <see cref="DataFrame"/> from an <see cref="IEnumerable"/> containing <see cref="GenericRow"/>s using the given schema.
+        /// It is important to make sure that the structure of every <see cref="GenericRow"/> of the provided <see cref="IEnumerable"/> matches
+        /// the provided schema. Otherwise, there will be runtime exception.
+        /// </summary>
+        /// <param name="data">List of Row objects</param>
+        /// <param name="schema">Schema as StructType</param>
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data, StructType schema) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Invoke( "createDataFrame",                 
+                data,      
+                DataType.FromJson(_jvmObject.Jvm, schema.Json)));
+
+        /// <summary>
+        /// Creates a dataframe from an <see cref="IEnumerable"/> containing <see cref="string"/> and no schema
+        /// </summary>
+        /// <param name="data">List of string objects</param>        
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<string> data)
+        {                       
+            var schemaFields = new List<StructField>();
+            schemaFields.Add(new StructField("_1", new StringType()));
+            var dataAsGenericRow = new List<GenericRow>();
+            foreach (string s in data)
+            {
+                dataAsGenericRow.Add(new GenericRow(new object[] {s}));
+            }            
+            return CreateDataFrame(dataAsGenericRow, new StructType(schemaFields));
+        }
+
+        /// <summary>
+        /// Creates a dataframe from an <see cref="IEnumerable"/> containing <see cref="int"/> and no schema
+        /// </summary>
+        /// <param name="data">List of int objects</param>        
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<int> data)
+        {
+            var schemaFields = new List<StructField>();
+            schemaFields.Add(new StructField("_1", new IntegerType()));
+            var dataAsGenericRow = new List<GenericRow>();
+            foreach (int i in data)
+            {
+                dataAsGenericRow.Add(new GenericRow(new object[] { i }));
+            }
+            return CreateDataFrame(dataAsGenericRow, new StructType(schemaFields));
```

Comment:
How about
```C#
private StructType SchemaWithSingleColumn(DataType dataType) =>
    new StructType(new List<StructField>() { new StructField("_1", dataType) });

private IEnumerable<GenericRow> ToGenericRows<T>(IEnumerable<T> rows) =>
    rows.Select(r => new GenericRow(new object[] { r }));
```

and use it like
```C#
public DataFrame CreateDataFrame(IEnumerable<int> data) =>
    CreateDataFrame(ToGenericRows(data),  SchemaWithSingleColumn(new IntegerType()));

public DataFrame CreateDataFrame(IEnumerable<string> data) =>
    CreateDataFrame(ToGenericRows(data),  SchemaWithSingleColumn(new StringType()));
```

Now, let's add all simple types since it's trivial to add one.